src/Pelagos/Util/DOIutil.php

```diff
@@ -375,4 +333,56 @@ function ($matches) {
             $input
         );
     }
+
+    /**
+     * This function generates a new DOI string that is not already in use per datacite.
+     *
+     * @return string A unique, unused DOI string.
+     */
+    public function generateDoi()
+    {
+        $encoder = new Base32CrockfordEncoder();
+        // 1099511627775 encodes to the longest 8 character Crockford 32 string.
+        $max = 1099511627775;
+        // Start at 1 (0 is problematic as library does not produce checksum for 0.)
+        $random = mt_rand(1, $max);
+        // Add shoulder and remove the checksum character.
+        $doi = $this->doishoulder . '/' . substr($encoder->encode($random), 0, -1);
+        while ($this->checkDoiExists($doi)) {
+            $random = mt_rand(0, $max);
```

Comment:
yep. good catch

Reply:
Also, the range (729299307643, 729299307645) can be used to force a frequent collision with 10.33516/n76t0k3x which I have put in as a draft-mode DOI on datacite's test system. (and maybe a couple dump statements to test)

Reply:
Also, to test non 400 conditions, I commented out the
#curl_setopt($ch, CURLOPT_URL, $this->url . '/id/' . $doi);
and used:
curl_setopt($ch, CURLOPT_URL, 'https://httpstat.us/404');

to insure the other errors still perculated through. This is otherwise difficult to test.